src/main/scala/io/iohk/ethereum/network/FastSyncActor.scala

```diff
@@ -0,0 +1,297 @@
+package io.iohk.ethereum.network
+
+import akka.actor.{Actor, ActorLogging, ActorRef, PoisonPill, Props}
+import akka.util.ByteString
+import io.iohk.ethereum.db.storage._
+import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.network.FastSyncActor._
+import io.iohk.ethereum.network.PeerActor.MessageReceived
+import io.iohk.ethereum.network.p2p.messages.PV62._
+import io.iohk.ethereum.network.p2p.messages.PV63._
+import io.iohk.ethereum.utils.Config.FastSync
+import org.spongycastle.util.encoders.Hex
+
+class FastSyncActor(
+  peerActor: ActorRef,
+  blockHeadersStorage: BlockHeadersStorage,
+  blockBodiesStorage: BlockBodiesStorage,
+  receiptStorage: ReceiptStorage,
+  mptNodeStorage: MptNodeStorage,
+  evmStorage: EvmCodeStorage) extends Actor with ActorLogging {
+
+  import context.{dispatcher, system}
+
+  val GenesisBlockNumber = 0
+  val EmptyAccountStorageHash = ByteString(Hex.decode("56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"))
+  val EmptyAccountEvmCodeHash = ByteString(Hex.decode("c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"))
+
+  override def receive: Receive = {
+    case StartSync(targetBlockHash) =>
+      //TODO check DB if it is not empty delete everything
+      peerActor ! PeerActor.Subscribe(Set(NodeData.code, Receipts.code, BlockBodies.code, BlockHeaders.code))
+      peerActor ! PeerActor.SendMessage(GetBlockHeaders(Right(targetBlockHash), 1, 0, reverse = false))
+      context become waitForTargetBlockHeader(targetBlockHash)
+  }
+
+  def waitForTargetBlockHeader(targetBlockHash: ByteString): Receive = {
+    case MessageReceived(BlockHeaders(blockHeaders)) if blockHeaders.nonEmpty =>
+      peerActor ! PeerActor.SendMessage(GetNodeData(Seq(blockHeaders.head.stateRoot)))
```

Comment:
I thin yes, because blockchain is easy to download from other pears and easy to resume download after failure.
This way we parallelize download and we are syncing faster